dao/authentication_dao.go

```diff
@@ -135,19 +135,22 @@ func (a *authenticationDaoImpl) ListForSource(sourceID int64, _, _ int, _ []util
 }
 
 func (a *authenticationDaoImpl) ListForApplication(applicationID int64, _, _ int, _ []util.Filter) ([]m.Authentication, int64, error) {
-	app := m.Application{ID: applicationID}
-	result := DB.
-		Where("tenant_id = ?", *a.TenantID).
-		Preload("ApplicationAuthentications").
-		First(&app)
-
-	if result.Error != nil {
-		return nil, 0, util.NewErrNotFound("application")
+	keys, err := a.listKeys()
```

Comment:
I believe we need first check, if given application ID exists ... something like in `ListForSource()`

Reply:
Changed on all of them - and fixed the Source one since it wasn't using tenancy.